platform/mds/mds/src/test/java/org/motechproject/mds/it/service/EntityServiceContextIT.java

```diff
@@ -215,6 +215,32 @@ public void shouldSaveEntityWithGivenLookups() throws IOException {
     }
 
     @Test
+    public void shouldAddNewFieldForLookupAndSaveEntity() throws IOException {
+        EntityDto entityDto = new EntityDto();
+        entityDto.setName("myEntity");
+        entityDto = entityService.createEntity(entityDto);
+
+        EntityDraft entityDraft = entityService.getEntityDraft(entityDto.getId());
+        LookupDto lookup = new LookupDto("lookup", false, false, null, true);
+        entityService.addLookups(entityDraft.getId(), Collections.singletonList(lookup));
+        entityService.saveDraftEntityChanges(entityDto.getId(), DraftBuilder.forNewField("disp", "testFieldName", Long.class.getName()));
+        FieldDto field = selectFirst(entityService.getFields(entityDto.getId()), having(on(FieldDto.class).getBasic().getName(), equalTo("testFieldName")));
+
+        Map<String, Object> values = new HashMap<>();
+        values.put("path", "indexes.0.$addField");
+        values.put("advanced", true);
+        values.put("value", Collections.singletonList(field.getId()));
+        DraftData draftData = new DraftData();
+        draftData.setEdit(true);
+        draftData.setValues(values);
+
+        entityService.saveDraftEntityChanges(entityDto.getId(), draftData);
+        entityService.commitChanges(entityDto.getId());
+
+        assertEquals("testFieldName", entityService.getLookupByName(entityDto.getId(),"lookup").getLookupField("testFieldName").getName());
```

Comment:
this assert is wrong because when this field doesn't exist it will be NPE here, please write assert in different way